api/api_post.py

```diff
@@ -22,4 +24,20 @@ def user_import():
         user.input_user()
         return user.name
     except Exception as e:
-        logging.exception(f"Exception [{e}]: while adding user")
\ No newline at end of file
+        logging.exception(f"Exception [{e}]: while adding user")
+
+@api_post.route('/api/add_song', methods=['POST'])
+@token_required
+def add_song():
+    try:
+        artist=request.args.get('artist')
+        genre=request.args.get('genre')
+        song_name=request.args.get('song_name')
+        yt_link=request.args.get('yt_link')
+        song = Song(artist, genre, song_name, yt_link)
+        song.input_song()
+        return song.song_name
+    except Exception as e:
+        logging.exception(f"Exception [{e}]: while adding song [{song.song_name}]")
```

Comment:
- We always have to return something from route methods - even if exception happens, check out `make_response` method from flask
- `param = request.args.get('param') is not safe - if i don't add that param, error message will look bad, and we will always return same error code and message for any error that happens (And all of the params here are required).
```
if 'artist' not in reques.args:
    return make_response("Artist field is required", 401)
artist=request.args.get('artist')
```
This way we can have more specific error codes and messages.

Also we should check for invalid parameters and return error there too, check the GET api file for reference on how to do it.

Reply:
Or if you want to do something more generalized , you can also say something like:
```
required_params = [ 'artist', 'song_name', 'genre', 'yt_link' ]

def add_song():
    ...
    for param in required_params:
        if param not in request.args:
            return make_response(f"{param} field is required", 401)

    artist=request.args.get('artist')
    genre=request.args.get('genre')
    song_name=request.args.get('song_name')
    yt_link=request.args.get('yt_link')
```